msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs

```diff
@@ -38,12 +39,18 @@ public override bool Execute ()
 				return false;
 			}
 
+			if (AssemblyName.IndexOf ('.') != AssemblyName.LastIndexOf ('.'))
+				LogAppManifestError ("The assembly name ({0}) cannot contain more than one dot.", AssemblyName);
```

Comment:
What happens if an assembly has more than one dot?

I ran into something similar for iOS, and added the same check, but it turned out lots of people have assemblies with more than one dot, so it became problematic. After some investigation it turned out that the problem wasn't more than one dot, but executables that ended with `.app`.